Ch foreign keys 160956347

mubstimor opened the pull request:
#### What does this PR do​?
Adds relationships across information in my database

#### Task to be completed
- POST /auth/signup
- POST /auth/login
- POST /users/orders
- GET /users/orders
- GET /orders
- GET ​/​orders​/<orderId>
- PUT /​orders​/<orderId>
- GET /menu
- POST /menu

#### How should this be manually tested?
Test endpoints​ with Postman by accessing https://tims-fast-food.herokuapp.com/api/v1/<endpoint>
_Endpoints are listed above._

#### Relevant pivotal tracker stories​
- #160956347
- #160788476

#### Screenshots
<img width="825" alt="screen shot 2018-10-03 at 23 36 41" src="https://user-images.githubusercontent.com/2491780/46459509-c9058d80-c7c1-11e8-8d72-3b87e3c1ef8f.png">

<img width="817" alt="screen shot 2018-10-04 at 10 32 51" src="https://user-images.githubusercontent.com/2491780/46459549-e89cb600-c7c1-11e8-8b2e-76dba971da53.png">

mubstimor committed:
fix(fastfoodfast): connect users, menu items to orders

Add foreign keys to connect orders to the menu

[Delivers #160956347]

mubstimor committed:
ch(fastfoodfast): Enforce PEP8 standards & test endpoints

start removing unused variables & clean code and ensuring endpoints
work as expected

[Delivers #160971278 #160876867]

mubstimor committed:
fix(fastfoodfast): run test for priority endpoints

Ensure priority endpoints work as required

[Delivers #160876867]

mubstimor committed:
fix(fastfoodfast): ensure all tests pass

Check all written tests are passing correctly

[Delivers #160876867]

mubstimor committed:
ch(fastfoodfast): Return human readable responses

Replace foreign keys in queries with appropriate identifier names

[Delivers #160956347]

mubstimor committed:
ch(fastfoodfast): improve code with pep8

Restructure code to match PEP8 guidelines

[Delivers 160980716]

mubstimor committed:
ch(fastfoodfast): update readme

Show which functionality is provided

[Delivers 160982299]

MuhweziDeo commented:
you can actually change your version to v2 to make it different from v1 of challenge 2

MuhweziDeo commented:
did you remove comments in code

mubstimor committed:
ch(fastfoodfast): update documentation

Add more routes to documentation page

[Delivers #160982299 #160848254]

mubstimor committed:
fix(fastfoodfast): add config file

Add config file that had been deleted

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): remove PEP8 violating comments

Remove comments that aren't upto standard

[Delivers #160982299 #160971278 #160980716]

mubstimor committed:
fix(fastfoodfast): update tests

update test parameter for order to satisy requirements

[Delivers #160876867 #160982299]

mubstimor committed:
fix(fastfoodfast): remove user id from order request

delete parameter from request as it's redundant

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): split views file into smaller files

Put related views in a given file

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): organised files into modules

Restructure app directory to improve readability

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): optimise code

remove unused lines of code

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): add missing module

add missing db module to models

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): reduce connections to db

use OOP to reduce number of connections

[Delivers #160982299]

mubstimor committed:
setup testdb on travis

mubstimor committed:
test bug fix close connect

mubstimor committed:
test bug fix close connect 2

mubstimor committed:
fix(fastfoodfast): all endpoints work

change queries to ensure endpoints are working & updated readme

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): update readme

add installation instructions

[Delivers #160982299]

mubstimor committed:
ch(fastfoodfast): point readme to recent badge

set repo to point to new coverage branch

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): use single connection

check if connection already exists before setting a new one

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): use connection pooling

open various threads for database connection

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): pass new test parameters

send new database test parameters

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): create tables in app module

create tables from a central location

[Delivers #160982299]

mubstimor committed:
fix(fastfoodfast): change cursor object

reinitialise cursor to db

[Delivers #160982299]

mubstimor committed:
print attribute error on heroku

mubstimor committed:
fix(fastfoodfast): close connections immediately

close each connection after a request

[Delivers #160982299]

mubstimor committed:
Merge pull request #13 from mubstimor/bg-heroku-deployment

Bg heroku deployment

mubstimor committed:
Merge pull request #14 from mubstimor/ch-presentation-version

Ch presentation version
